platforms/documentation/docs/src/docs/userguide/reference/best-practices/best_practices_tasks.adoc

```diff
@@ -409,3 +409,84 @@ The output reveals that the `lib` project is now built when the `:app:goodClassp
 
 === Tags
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:inputs-and-outputs,#inputs-and-outputs>>`, `<<tags_reference.adoc#tag:configurations,#configurations>>`
+
+[[use_unique_output_files_and_directories]]
+== Use unique output files and directories
+
+Overlapping output files or directories cause tasks to rerun unnecessarily and waste work.
+
+=== Explanation
+
+Gradle tracks all output files and directories declared by tasks to decide whether a task needs to be rerun.
+For example, if the contents of a task’s output directory change after its last execution, Gradle will rerun that task.
+
+Ensuring that each task uses its own unique output files and directories, both within a project and across the entire build, prevents unnecessary work.
+
+=== Example
+
+==== Don't Do This
+
+====
+include::sample[dir="snippets/bestPractices/uniqueOutputs-avoid/kotlin",files="build.gradle.kts[tags=avoid-this]"]
+include::sample[dir="snippets/bestPractices/uniqueOutputs-avoid/groovy",files="build.gradle[tags=avoid-this]"]
+====
+
+<1> *Write to a file in the output directory*: This task produces a single file in the `outputDirectory`, named based on the `type` input.
+<2> *Read a specific file in the input directory*: This task only needs to read a single `a.txt` file in the input directory.
+<3> *Set output directory*: Sets `outputDirectory` to a subdirectory in `buildDirectory`.
+<4> *Set output directory*: Same as above, using the same shared `greetings` directory.
+<5> *Wire `greeterA` to consumer*: Makes sure that `greeterA` runs and produces the output directory before it is used by `consumer`.
+
+With this setup, if you run the `consumer` task, then `greeterB`, the `consumer` task will be invalidated.
+
+The next time `consumer` is run it will *not* be `UP-TO-DATE` and will have to run again despite not using the output from `greeterB`.
```

Comment:
```suggestion
The next time you run `consumer`, it will *not* be `UP-TO-DATE` and will execute again, even though it does not use the output from `greeterB`.
```